services/_shared/github_rulesets_client.py

```
@@ -326,18 +351,32 @@ def detect_enforcement(
-        if not _check_name_in_ruleset(rs, check_name):
+        if rs.get("enforcement") != "active" or rs.get("target") != "branch":
+            continue
+        full = get_ruleset(install_token, owner, repo, rs["id"])
+        if not _check_name_in_ruleset(full, check_name):
             continue
         if stored_ruleset_id is not None and rs.get("id") == stored_ruleset_id:
             grug_match = True
+            break
         elif rs.get("name", "").startswith(GRUG_RULESET_PREFIX):
             grug_match = True
+            break
```

Ruff (0.15.20)

[warning] Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)